syfertext/doc.py

```diff
@@ -217,6 +222,87 @@ def similarity(self, other: "Doc") -> torch.Tensor:
 
         return sim
 
+    @property
+    def is_sentenced(self):
+        """Check if the document has sentence boundaries assigned. This is
+        defined as having at least one token other than the first where sent_start is not None.
+        """
+
+        # check for empty doc
+        if len(self) == 0:
+            return False
+
+        return (self.container[0].is_sent_start != None) and (self.container[0].is_sent_start)
+
+    @property
+    def sents(self):
+        """Iterate over the sentences in the document. Yields sentence `Span`
+        objects.
+
+        Returns:
+            List of Spans: Sentences in the document.
+        """
+        assert (
+            self.is_sentenced
+        ), "Doc has not been sentencised yet, please use the sentenciser pipeline to divide doc into senetences"
+
+        start = 0
+        for i in range(1, len(self)):
+            if self.container[i].is_sent_start:
+                span = Span(self, start, i, owner=self.owner)
+                span.client_id = self.client_id
+                start = i
+                yield span
+
+        if start != len(self):
+            span = Span(self, start, len(self), owner=self.owner)
+            span.client_id = self.client_id
+            yield span
+
+    def _get_a_sentence(self):
```

Comment:
Please complete this docstring with type hinting of the returned value and a `Returns:` section